### PR TITLE
Fix DNS config, expired certificates, and missing requirements

### LINKED ISSUE #92: How does company router install packerbeat?
Hi

I want to install new application in hosts, however it seems that there is no any dns configuration in all host of socbed, which means they are not capable to resolve any domain name, but how does this ansible script work:
```
- name: Download packetbeat to /tmp
  get_url:
    url: "https://artifacts.elastic.co/downloads/beats/packetbeat/{{ packetbeat_name }}.tar.gz"
    dest: "/tmp"
  retries: 2
  register: result
  until: result is succeeded
```
by the way, I find that the gateway of internet router disappear sometimes， do you have any idea of this?

if my memory serves me right, all hosts can visit external network by ip instead of domain because internet router have a network interface which is nat mode, and none of them configure dns right?

thanks

### LINKED ISSUE #93: Missing dependencies in Attacker Server
**Describe the bug**
Missing two dependencies in Attack Search:
* missing pip dependencies of aiosmptd when updating tests after switching from smptd to aiosmtpd in Attacker
* missing systemd-resolved apt in Attacker Server so that the `/run/systemd/resolv/resolv.conf` is not exist

**To Reproduce**
For the first missing dependency:  running `/usr/bin/python3 /usr/bin/external_mail_handler.py`, then the failure is as below:
image
And after install python3-aiosmtpd, we can running `/usr/bin/python3 /usr/bin/external_mail_handler.py` successfully, however, the service still can not running successfully with error as below:
image

For the second: running `ls /run/systemd/`, the result doesn't have a directory named `resolv`

**Expected behavior**
* install python3-aiosmtpd in Attacker using apt, and change the `input('SMTP server running. Press Return to stop server and exit.')` to below code:
```python
    try:
        while True:
            pass  # keep running
    except KeyboardInterrupt:
        self.controller.stop()
```
* install systemd-resolved after attacker is installed using apt

### PR DESCRIPTION
**Pull request by Maspital**
This addresses a couple of small issues:
- The attacker did not have internet access post-provisioning due to missing DNS config
  - I opted to simply set it directly via `/etc/resolv.conf`, aka, without systemd-resolved
- The attacker couldn't access package repositories due to expired certificates
  - Fixed by adding the current keychain during the post-boot command
- The external mail handler running on the attacker is missing the aiosmtpd package
  - This change is recent, I simply forgot to also add this packet when updating from the deprecated smtpd
  - After fixing this, I've noticed that using `input()` within a service isn't a good idea, so I changed the service to simply run forever. It can still be restarted etc. with `systemctl`
- Sometimes, the Internet Router is unable to provide internet access to the rest of the network (and itself) for a short period of time
  - Bumping to the latest 2.x core version does not solve this
  - Seems to work reliably when explicitly setting DNS in `/var/ipfire/dns/servers` (I chose quad8)
  - On the same note, while upgrading from core141 to 157 on the internet router works without problems, it messes up the provisioning on the company router by preventing ansible from reconnecting after a restart (see [here](https://forum.ansible.com/t/reboot-task-no-longer-reconnects-or-times-out/10483)). We could still upgrade by running `pakfire update`/`pakfire upgrade` as the last provisioning step, but this takes quite a while, so I've opted to leave it at 141.

Resolves #93 
Resolves #92 
Also related to the mail handler issue mentioned in #80, but no longer relevant as the "buggy" part of the code has been replaced when switching to `aiosmtpd`

### CODE REVIEW

File: provisioning/ansible/roles/external_mail_handler/tasks/main.yml [hunk 1]
```diff
@@ -4,6 +4,18 @@
 # Additionally, creates a logrotate config file for logs
 #
 
+- name: Upgrade binutils
```
Comment (ru37z) [inline provisioning/ansible/roles/external_mail_handler/tasks/main.yml:7]: A principal idea of SOCBED is to pin as many package versions as possible to (1) ensure reproducibility and (2) avoid unforeseen problems in the future. I'm actually not sure how many packages and other software are currently affected by this, we should discuss whether is makes sense to pin for example this package version here.

**Review by ru37z — CHANGES_REQUESTED** [review]
Discussion required

**Review by ru37z — APPROVED** [review]

**Comment by 1ncludeSteven** [comment]
If you want to upgrade company router from IPFire2.25 to IPFire2.29, then current packetbeat version can not be executed correctly with error message `runtime/cgo: pthread_create failed: Operation not permitted
`

**Comment by Maspital** [comment]
> If you want to upgrade company router from IPFire2.25 to IPFire2.29, then current packetbeat version can not be executed correctly with error message `runtime/cgo: pthread_create failed: Operation not permitted `

We've decided to not upgrade the company router for now since the current older version works fine. But thank you for the heads-up!

**Pull request merged**